### PR TITLE
Allow `-target` to be specified on the command line. NFC

### PR DESCRIPTION
**Pull request by sbc100**
This makes `-target wasm64` work and imply `-sMEMORY64`.

Supporting these flags makes us more compatible with upstream clang and with wasi-sdk.

See #19834

### CODE REVIEW

File: emcc.py [hunk 1]
```diff
@@ -1691,15 +1700,6 @@ def setup_pthreads(target):
     'emscripten_stack_set_limits',
   ]
 
-  if settings.MAIN_MODULE:
```
Comment (dschuff) [inline emcc.py:1694]: is this related?

File: emcc.py [hunk 2]
```diff
@@ -3637,6 +3632,10 @@ def consume_arg_file():
     elif arg.startswith('-o'):
       options.output_file = removeprefix(arg, '-o')
       newargs[i] = ''
+    elif check_arg('-target') or check_arg('--target'):
```
Comment (dschuff) [inline emcc.py:3635]: Clang doesn't seem to support `--target`, only `-target`. Maybe we should do the same?

**Review by dschuff — COMMENTED** [review]
Should we also support `-m32` / `-m64`?

File: emcc.py (hunk 2 above)
Comment (sbc100) [reply emcc.py:3635]: clang supports both `--target=wasm32` and `-target` `wasm32`

File: emcc.py (hunk 1 above)
Comment (sbc100) [reply emcc.py:1694]: Oops.. different change.

File: emcc.py (hunk 2 above)
Comment (dschuff) [reply emcc.py:3649]: oh, and we've already separated out the `=` delimiter here?

**Review by dschuff — APPROVED** [review]
